### DISCUSSION TITLE
Split GPX imported segments in differents tracks

### DISCUSSION THREAD

**Discussion by YvesDen** [osmandapp/OsmAnd#20612] [General]
Hello,

I use OSMAND to navigate in Southwest. 
I create a lot of tracks under Basecamp, because I use hiking tracks on a Garmin handheld, and 4x4 tracks on my phone.
Is there a way to split a GPX containing different segments into different tracks ? I would like to change some colors of some tracks. 

The way I do it : a file on Basecamp is containing all tracks (another all points, because if you try to import a GPX containing tracks AND points it does not work properly). I export this file in GX format from Basecamp to my desktop. I import it on the phone, and into OSMAND. it creates a single track, containing different "segments". I can not split. 

The only way is to export each tracks from Basecamp separately, but it take too much time and effort. 

Thanks

**Comment by tmusolf** [comment]
I mention this in case it's of help with your similar issue.   

I use google maps to create custom maps that I download into OSMAnd and I had a similar problem.  I created a utility [GoogleMapsToOSMAndGPX ](https://github.com/tmusolf/GoogleMapToOSMAndGPX )to convert the google map data into OSMAnd compatible GPX files which can then be copied into the appropriate directory on your phone.  It does icon conversion, separates tracks into individual files preserving line color and width, etc.

**Reply by YvesDen** [reply]
Hello,

thanks.
But I don't see the app or software link to download it.
Or the way to use it.

Yours,

**Reply by tmusolf** [reply]
Were you able to follow the link in my previous reply to the GitHub repository for GoogleMapsToOSMAndGPX ? That's were the code and documentation are located.

**Reply by YvesDen** [reply]
yes I followed, but "the cade" ? Lol, I'm not programer...

**Reply by tmusolf** [reply]
To find the windows programs (assuming you're running windows) that **don't** require you to use python or deal with any code go to the GitHub page that I linked to.  Inside the **dist** folder you'll find two windows programs.  

Download them both and put them both in the same folder.  You don't have to install them.  Run the one that ends in **-gui**  That will launch the gui wrapper described in the documentation.

This was a quick program I put together to solve my issue.  Over a few versions I've cleaned it up, added some more options, wrote the gui wrapper, created the windows executables and made a decent first pass at documentation in the hope that it will be useful to others in a similar situation.  I use it all the time and it works well for me.  

Hope that helps.

**Reply by YvesDen** [reply]
Tried to exectute, but does not want on my desktop. 
Win 10 64b

**Reply by tmusolf** [reply]
So when you double click on GoogleMapToOSMAndGPX-gui.exe does just nothing happen? If that's the case I don't know what's going on.  It runs on my various systems.
If windows puts up a message about unknown publisher/unrecognized app you just have to tell it to run anyway.  I don't have a official windows developer certificate so I can't "sign" my apps to make them official.

**Reply by YvesDen** [reply]
First time box to run it anyway, then box does not ru on this system...

**Reply by tmusolf** [reply]
Not sure what to say.  It runs on my 3 windows 10 systems.  Both exe's need to be in the same directory and make sure you click on the GUI one.  If you try and run the other one by clicking on it it won't do anything.
![image](https://github.com/user-attachments/assets/4ef63726-d4f2-4dde-be4a-8cfc378bd0a1)
Sorry I can't offer up any other suggestions.

**Reply by YvesDen** [reply]
both are in the same directory.

here is the message. 


![Capturer_29](https://github.com/user-attachments/assets/661b81fc-412d-4f23-835c-dbcfa61870d0)

**Comment by tmusolf** [comment]
Looking at the message you sent I had one other thought.  Windows and applications can be built to be either 32 or 64 bit.  Most all windows installs these days are 64 bit, but there may still be some older systems that are running 32 bit versions. 

I built the exe's as 64 bit programs, which means that they will only run on 64 bit versions of Windows.  If your system is 32 bit then the exe's won't run and will display that message.  

To check if your windows version is 32 or 64 bit follow these steps.

    Press the Start button, then select Settings.
    Click on System.
    Go to About and look for the System type field to check which version you have.
![image](https://github.com/user-attachments/assets/4d9b0840-6fc4-44c9-abb1-d31d39636930)

**Reply by YvesDen** [reply]
It's 64b, I told you first answer.
